src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -312,17 +317,10 @@ private void AssertValidConnection(IncomingConnectionInfo connectionInfo)
                     $"Cannot accept the incoming replication connection from {connectionInfo.SourceUrl}, because this node is in passive state.");
             }
 
-            if (_incoming.TryRemove(connectionInfo.SourceDatabaseId, out IncomingReplicationHandler value))
+            if (_incoming.TryGetValue(connectionInfo.SourceDatabaseId, out var value))
             {
-                if (_log.IsInfoEnabled)
-                {
-                    _log.Info(
-                        $"Disconnecting existing connection from {value.FromToString} because we got a new connection from the same source db");
-                }
-
-                IncomingReplicationRemoved?.Invoke(value);
-
-                value.Dispose();
+                throw new InvalidOperationException(
+                    $"connection for this database already exists from {value.ConnectionInfo.SourceUrl}.");
```

Comment:
Minor: connection -> Connection

Reply:
I think we want to make this slightly smarter. Do we have any state around the connection?
Can we see when was the last time we got something from it?

Reply:
We can have the last heartbeat time and last etag from source (and maybe other stuff as well).
Do we want here a better exception? or we want to decide which connection should be dropped?

Reply:
We always want better errors :-)
But yes, if the old connection hasn't had communication recently, we want to drop that and let the new one in